src/main/java/org/folio/linked/data/integration/kafka/sender/search/WorkCreateMessageSender.java

```diff
@@ -39,36 +37,29 @@ public class WorkCreateMessageSender implements CreateMessageSender {
   private final FolioMessageProducer<ResourceIndexEvent> bibliographicMessageProducer;
   private final KafkaSearchMessageMapper<LinkedDataWork> searchBibliographicMessageMapper;
   private final ApplicationEventPublisher eventPublisher;
-  private final ResourceRepository resourceRepository;
 
   @Override
   public Collection<Resource> apply(Resource resource) {
-    return Stream.of(resource)
-      .filter(this::test)
-      .toList();
+    if (resource.isOfType(WORK)) {
+      return singletonList(resource);
+    }
+    if (resource.isOfType(INSTANCE)) {
+      triggerParentWorkUpdate(resource);
+    }
+    return emptyList();
   }
 
-  //TODO refactoring to extract resources
-  private boolean test(Resource resource) {
-    return ResourceKind.BIBLIOGRAPHIC
-      .stream()
-      .anyMatch(resource::isOfType);
+  private void triggerParentWorkUpdate(Resource instance) {
+    extractWork(instance)
+      .ifPresent(work -> {
+        log.info("Instance [id {}] creation triggered parent Work [id {}] update", instance.getId(), work.getId());
+        eventPublisher.publishEvent(new ResourceUpdatedEvent(work, null));
```

Comment:
I think having oldWork as "null" here will cause a null pointer exception in line 116 of WorkUpdateMessageHandler line 116. (`eventPublisher.publishEvent(new ResourceDeletedEvent(oldResource));` --> oldResource will be null and hence NPE)

Reply:
that's true, thank you for the notice! fixed